liquidctl/driver/msi.py

```diff
@@ -0,0 +1,1076 @@
+    "disable2":                     _LightingMode.DISABLE2,
+    "double flashing":              _LightingMode.DOUBLE_FLASHING,
+    "double meteor":                _LightingMode.DOUBLE_METEOR,
+    "energy":                       _LightingMode.ENERGY,
+    "fan control":                  _LightingMode.FAN_CONTROL,
+    "fire":                         _LightingMode.FIRE,
+    "flashing":                     _LightingMode.FLASHING,
+    "jazz":                         _LightingMode.JAZZ,
+    "jrainbow":                     _LightingMode.JRAINBOW,
+    "lava":                         _LightingMode.LAVA,
+    "lightning":                    _LightingMode.LIGHTNING,
+    "marquee":                      _LightingMode.MARQUEE,
+    "meteor":                       _LightingMode.METEOR,
+    "movie":                        _LightingMode.MOVIE,
+    "msi marquee":                  _LightingMode.MSI_MARQUEE,
+    "msi rainbow":                  _LightingMode.MSI_RAINBOW,
+    "steady":                       _LightingMode.NO_ANIMATION,
+    "planetary":                    _LightingMode.PLANETARY,
+    "play":                         _LightingMode.PLAY,
+    "pop":                          _LightingMode.POP,
+    "rainbow":                      _LightingMode.RAINBOW,
+    "rainbow double flashing":      _LightingMode.RAINBOW_DOUBLE_FLASHING,
+    "rainbow flashing":             _LightingMode.RAINBOW_FLASHING,
+    "rainbow wave":                 _LightingMode.RAINBOW_WAVE,
+    "random":                       _LightingMode.RANDOM,
+    "rap":                          _LightingMode.RAP,
+    "stack":                        _LightingMode.STACK,
+    "visor":                        _LightingMode.VISOR,
+    "water drop":                   _LightingMode.WATER_DROP,
+    "end":                          _LightingMode.END,
+    }
+
+    _MATCHES = [
+        (0x0db0, 0xb130, 'MSI MPG Coreliquid K360', {
+            'fan_count': 5
+        }),
+        (0x0db0, 0xca00, 'Unknown', {}),
+        (0x0db0, 0xca02, 'Unknown', {}),
+    ]
+
+    def __init__(self, device, description, **kwargs):
+        super().__init__(device, description, **kwargs)
+        self._feature_data_per_led = bytearray(_PER_LED_LENGTH + 5)
+        self._bytearray_oled_hardware_monitor_data = bytearray(_REPORT_LENGTH)
+        self._per_led_rgb_jonboard = bytearray(_PER_LED_LENGTH)
+        self._per_led_rgb_jrainbow1 = bytearray(_PER_LED_LENGTH)
+        self._per_led_rgb_jrainbow2 = bytearray(_PER_LED_LENGTH)
+        self._per_led_rgb_jcorsair = bytearray(_PER_LED_LENGTH)
+        self._fan_count = kwargs.pop('fan_count', 5)
+        # the following fields are only initialized in connect()
+        self._data = None
+        self._feature_data = None
+
+    @classmethod
+    def probe(cls, handle, **kwargs):
+        if (handle.hidinfo['usage_page'] != USAGE_PAGE):
+            return
+        yield from super().probe(handle, **kwargs)        
```

Comment:
Please keep the same comments as in `RgbFusion2.probe`, since this isn't a trivial piece of code.

Reply:
Fixed by https://github.com/liquidctl/liquidctl/pull/564/commits/c4ac9a2677690e84851b1ab642e8d0efa6feeeb9 and https://github.com/liquidctl/liquidctl/pull/564/commits/5282e7776d9cb3299e916c29e26312c9dd5fc2da together